### PR TITLE
#278. Upgrade go-oscal package. 

### LINKED ISSUE #278: Update Lula to using the latest version of go-oscal
Upon release of go-oscal v0.2.0 we will want to update Lula.

## Purpose
Types have changed and updating the output of Lula to produce oscal version `1.1.2` documents as aligned with the latest release of OSCAL is our goal. 

## Objectives
- Any modifications to data types used (Should only be `component-definition` and `assessment-results` at this time)
- Update the `OSCAL_VERSION` constant for `assessment-results`
- Update component-definitions in the e2e tests to `1.1.2`

### PR DESCRIPTION
**Pull request by mike-winberry**
Upon release of go-oscal v0.2.0 we will want to update Lula.

## Purpose
Types have changed and updating the output of Lula to produce oscal version `1.1.2` documents as aligned with the latest release of OSCAL is our goal. 

## Objectives
- Any modifications to data types used (Should only be `component-definition` and `assessment-results` at this time)
- Update the `OSCAL_VERSION` constant for `assessment-results`
- Update component-definitions in the e2e tests to `1.1.2`

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (brandtkeller) [inline .vscode/settings.json]: Is this intended to be included in the PR?

Comment (mike-winberry) [reply .vscode/settings.json]: nope must not be part of the ignore.

Comment (mike-winberry) [reply .vscode/settings.json]: I removed and added to .gitignore.

**Review by brandtkeller — APPROVED** [review]
I don't see any outliers as far as the types go 👍

**Pull request merged**